### PR TITLE
frame_write: Use tokio_util::io::framed_write

### PR DESCRIPTION
**Pull request by jxs**

### CODE REVIEW

**Review by seanmonstar — APPROVED** [review]
Looks good to me, assuming that the util function does essentially the same thing. I tagged Eliza since I see a bunch of her comments being deleted, but :shrug:

**Review by hawkw — APPROVED** [review]
looks good to me --- as you can see, i *definitely* followed up on that TODO as soon as `tokio-util` 0.5.1 was released... :innocent:

**Comment by hawkw** [comment]
@seanmonstar 
> [...] assuming that the util function does essentially the same thing [...]

I'm pretty sure it does, because IIRC, I basically copied and pasted the (at the time unpublished) code directly out of `tokio-util`...

File: src/codec/framed_write.rs [hunk 1]
```diff
@@ -126,21 +121,11 @@ where
...
                     }
                     _ => {
                         tracing::trace!(queued_data_frame = false);
-                        ready!(write(
-                            &mut self.inner,
-                            self.encoder.is_write_vectored,
-                            &mut self.encoder.buf,
-                            cx,
-                        ))?
+                        ready!(write(&mut self.inner, &mut self.encoder.buf, cx,))?
```
Comment (hawkw) [inline src/codec/framed_write.rs:128]: similarly, if we got rid of the `write` fn, we could just call `poll_write_buf` here:
```suggestion
                        ready!(tokio_util::io::poll_write_buf(Pin::new(&mut self.inner), &mut buf, cx))?
```

File: src/codec/framed_write.rs [hunk 2]
```diff
@@ -126,21 +121,11 @@ where
                     Some(Next::Data(ref mut frame)) => {
                         tracing::trace!(queued_data_frame = true);
                         let mut buf = (&mut self.encoder.buf).chain(frame.payload_mut());
-                        ready!(write(
-                            &mut self.inner,
-                            self.encoder.is_write_vectored,
-                            &mut buf,
-                            cx,
-                        ))?
+                        ready!(write(&mut self.inner, &mut buf, cx,))?
```
[the remark this answers is not in the document]
Comment (jxs) [reply src/codec/framed_write.rs]: thanks Eliza! yeah makes total sense, updated!

**Review by hawkw — APPROVED** [review]

**Comment by LucioFranco** [comment]
Shall we merge?

**Comment by jxs** [comment]
superseeded by #712

**Pull request closed without merging**